File: Data Structures & Algorithms/restore-ip-addresses/submission-0.py
class Solution:
    def restoreIpAddresses(self, s: str) -> List[str]:
        i = 0
        j = 1
        k = 2
        l = 3
        n = len(s)
        places = n - 1

        res = []
        for i in range(n-1):
            for j in range(i+1, n-1):
                for k in range(j+1, n-1):
                        print(s[:i+1], s[i+1:j+1], s[j+1:k+1], s[k+1:])
                        a = s[:i+1]
                        b = s[i+1:j+1]
                        c = s[j+1:k+1]
                        d = s[k+1:]
                        if int(a) > 255 or int(b) > 255 or int(c) > 255 or int(d) > 255:
                            continue
                        elif (len(a) > 1 and a[0] == "0"):
                            continue
                        elif (len(b) > 1 and b[0] == "0"):
                            continue
                        elif (len(c) > 1 and c[0] == "0"):
                            continue
                        elif (len(d) > 1 and d[0] == "0"):
                            continue
                        else:
                            res.append(a+"."+b+"."+c+"."+d)
        return res

File: Data Structures & Algorithms/same-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool res = true;
    void traverse(TreeNode* p, TreeNode* q){
        if(p==NULL && q==NULL) return;
        if(p==NULL && q!=NULL) {
            res = false;
            return;
        }
        if(p!=NULL && q==NULL) {
            res = false;
            return;
        }
        traverse(p->left, q->left);
        std::cout << p->val << " " << q->val << std::endl;
        if(p->val != q->val) res =false;
        traverse(p->right, q->right);
    }
    bool isSameTree(TreeNode* p, TreeNode* q) {
        traverse(p, q);
        return res;
    }
};
